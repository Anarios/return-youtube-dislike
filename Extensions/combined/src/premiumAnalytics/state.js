export const RANGE_OPTIONS = [7, 30, 90, 0];
export const RANGE_ANCHORS = ["first", "last"];
export const COUNTRY_LIMIT = 12;

export const analyticsState = {
  initialized: false,
  panelElement: null,
  panelExpanded: false,
  isLoading: false,
  activityChart: null,
  mapChart: null,
  currentVideoId: null,
  currentRange: 30,
  rangeAnchor: "first",
  currentMode: "ratio",
  activeRequestKey: null,
  latestCountries: [],
  latestSubdivisions: [],
  sessionToken: null,
  sessionActive: false,
  membershipTier: "none",
  latestSeriesPoints: [],
  latestTimeAxis: [],
  latestBucketMs: 60 * 60 * 1000,
  latestBucketLabel: null,
  chartTimeBounds: { min: null, max: null },
  globalTimeBounds: { min: null, max: null },
  availableRange: { min: null, max: null },
  selectionRange: { from: null, to: null },
  customSelection: null,
  suppressZoomEvents: false,
  zoomListenerRegistered: false,
  usingCustomRange: false,
  pendingSelection: null,
  expandedChart: null,
  mapView: "world",
  mapFocusCountry: null,
};

export function resetStateForVideo() {
  analyticsState.latestSeriesPoints = [];
  analyticsState.latestTimeAxis = [];
  analyticsState.latestBucketMs = 60 * 60 * 1000;
  analyticsState.latestBucketLabel = null;
  analyticsState.chartTimeBounds = { min: null, max: null };
  analyticsState.globalTimeBounds = { min: null, max: null };
  analyticsState.availableRange = { min: null, max: null };
  analyticsState.selectionRange = { from: null, to: null };
  analyticsState.customSelection = null;
  analyticsState.suppressZoomEvents = false;
  analyticsState.pendingSelection = null;
  analyticsState.usingCustomRange = false;
  analyticsState.expandedChart = null;
  analyticsState.latestCountries = [];
  analyticsState.latestSubdivisions = [];
  analyticsState.mapView = "world";
  analyticsState.mapFocusCountry = null;
}

export function resetSessionState() {
  analyticsState.sessionToken = null;
  analyticsState.sessionActive = false;
  analyticsState.membershipTier = "none";
}

export function setSession(token, isActive, membershipTier = "none") {
  analyticsState.sessionToken = token;
  analyticsState.sessionActive = !!isActive;
  analyticsState.membershipTier = typeof membershipTier === "string" ? membershipTier : "none";
}
