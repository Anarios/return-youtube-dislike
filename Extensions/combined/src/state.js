import {
  getLikeButton,
  getDislikeButton,
  getButtons,
  getLikeTextContainer,
  getDislikeTextContainer,
} from "./buttons";
import { createRateBar } from "./bar";
import {
  getBrowser,
  getVideoId,
  cLog,
  numberFormat,
  getColorFromTheme,
} from "./utils";
import { localize } from "./utils";
import { createStarRating } from "./starRating";

//TODO: Do not duplicate here and in ryd.background.js
const apiUrl = "https://returnyoutubedislikeapi.com";
const LIKED_STATE = "LIKED_STATE";
const DISLIKED_STATE = "DISLIKED_STATE";
const NEUTRAL_STATE = "NEUTRAL_STATE";

let extConfig = {
  disableVoteSubmission: false,
  coloredThumbs: false,
  coloredBar: false,
  colorTheme: "classic",
  numberDisplayFormat: "compactShort",
  showTooltipPercentage: false,
  tooltipPercentageMode: "dash_like",
  numberDisplayReformatLikes: false,
};

let storedData = {
  likes: 0,
  dislikes: 0,
  previousState: NEUTRAL_STATE,
};

function isMobile() {
  return location.hostname == "m.youtube.com";
}

function isShorts() {
  return location.pathname.startsWith("/shorts");
}

function isNewDesign() {
  return document.getElementById("comment-teaser") !== null;
}

function isRoundedDesign() {
  return document.getElementById("segmented-like-button") !== null;
}

let mutationObserver = new Object();

if (isShorts() && mutationObserver.exists !== true) {
  cLog("initializing mutation observer");
  mutationObserver.options = {
    childList: false,
    attributes: true,
    subtree: false,
  };
  mutationObserver.exists = true;
  mutationObserver.observer = new MutationObserver(function (
    mutationList,
    observer
  ) {
    mutationList.forEach((mutation) => {
      if (
        mutation.type === "attributes" &&
        mutation.target.nodeName === "TP-YT-PAPER-BUTTON" &&
        mutation.target.id === "button"
      ) {
        // cLog('Short thumb button status changed');
        if (mutation.target.getAttribute("aria-pressed") === "true") {
          mutation.target.style.color =
            mutation.target.parentElement.parentElement.id === "like-button"
              ? getColorFromTheme(true)
              : getColorFromTheme(false);
        } else {
          mutation.target.style.color = "unset";
        }
        return;
      }
      cLog(
        "unexpected mutation observer event: " + mutation.target + mutation.type
      );
    });
  });
}

function isLikesDisabled() {
  // return true if the like button's text doesn't contain any number
  if (isMobile()) {
    return /^\D*$/.test(
      getButtons().children[0].querySelector(".button-renderer-text").innerText
    );
  }
  return /^\D*$/.test(getButtons().children[0].innerText);
}

function isVideoLiked() {
  if (isMobile()) {
    return (
      getLikeButton().querySelector("button").getAttribute("aria-label") === "true"
    );
  }
  return getLikeButton().classList.contains("style-default-active") || getLikeButton().querySelector('button')?.getAttribute('aria-pressed') === 'true';
}

function isVideoDisliked() {
  if (isMobile()) {
    return (
      getDislikeButton().querySelector("button").getAttribute("aria-label") === "true"
    );
  }
  return getDislikeButton().classList.contains("style-default-active") || getDislikeButton().querySelector('button')?.getAttribute('aria-pressed') === 'true';
}

function getState(storedData) {
  if (isVideoLiked()) {
    return { current: LIKED_STATE, previous: storedData.previousState };
  }
  if (isVideoDisliked()) {
    return { current: DISLIKED_STATE, previous: storedData.previousState };
  }
  return { current: NEUTRAL_STATE, previous: storedData.previousState };
}

//---   Sets The Likes And Dislikes Values   ---//
function setLikes(likesCount) {
  cLog(`SET likes ${likesCount}`)
  getLikeTextContainer().innerText = likesCount;
}

function setDislikes(dislikesCount) {
  cLog(`SET dislikes ${dislikesCount}`)
  getDislikeTextContainer()?.removeAttribute("is-empty");
  getDislikeTextContainer()?.removeAttribute('is-empty');
  if (!isLikesDisabled()) {
    if (isMobile()) {
      getButtons().children[1].querySelector(
        ".button-renderer-text"
      ).innerText = dislikesCount;
      return;
    }
    getDislikeTextContainer().innerText = dislikesCount;

  } else {
    cLog("likes count disabled by creator");
    if (isMobile()) {
      getButtons().children[1].querySelector(
        ".button-renderer-text"
      ).innerText = localize("TextLikesDisabled");
      return;
    }
    getDislikeTextContainer().innerText = localize("TextLikesDisabled");
  }
}

function getLikeCountFromButton() {
  try {
    if (isShorts()) {
      //Youtube Shorts don't work with this query. It's not necessary; we can skip it and still see the results.
      //It should be possible to fix this function, but it's not critical to showing the dislike count.
      return false;
    }

    let likeButton = getLikeButton()
    .querySelector("yt-formatted-string#text") ??
    getLikeButton().querySelector("button");

    let likesStr = likeButton.getAttribute("aria-label")
    .replace(/\D/g, "");
    return likesStr.length > 0 ? parseInt(likesStr) : false;
  } catch {
    return false;
  }
}

function processResponse(response, storedData) {
  const formattedDislike = numberFormat(response.dislikes);
  setDislikes(formattedDislike);
  if (extConfig.numberDisplayReformatLikes === true) {
    const nativeLikes = getLikeCountFromButton();
    if (nativeLikes !== false) {
      setLikes(numberFormat(nativeLikes));
    }
  }
  storedData.dislikes = parseInt(response.dislikes);
  storedData.likes = getLikeCountFromButton() || parseInt(response.likes);
  createRateBar(storedData.likes, storedData.dislikes);
  if (extConfig.coloredThumbs === true) {
    if (isShorts()) {
      // for shorts, leave deactivated buttons in default color
      let shortLikeButton = getLikeButton().querySelector(
        "tp-yt-paper-button#button"
      );
      let shortDislikeButton = getDislikeButton().querySelector(
        "tp-yt-paper-button#button"
      );
      if (shortLikeButton.getAttribute("aria-pressed") === "true") {
        shortLikeButton.style.color = getColorFromTheme(true);
      }
      if (shortDislikeButton.getAttribute("aria-pressed") === "true") {
        shortDislikeButton.style.color = getColorFromTheme(false);
      }
      mutationObserver.observer.observe(
        shortLikeButton,
        mutationObserver.options
      );
      mutationObserver.observer.observe(
        shortDislikeButton,
        mutationObserver.options
      );
    } else {
      getLikeButton().style.color = getColorFromTheme(true);
      getDislikeButton().style.color = getColorFromTheme(false);
    }
  }
  //Temporary disabling this - it breaks all places where getButtons()[1] is used
  // createStarRating(response.rating, isMobile());
}

// Tells the user if the API is down
function displayError(error) {
  getDislikeTextContainer().innerText = localize(
    "textTempUnavailable"
  );
}

async function setState(storedData) {
  storedData.previousState = isVideoDisliked()
    ? DISLIKED_STATE
    : isVideoLiked()
    ? LIKED_STATE
    : NEUTRAL_STATE;
  let statsSet = false;

  let videoId = getVideoId(window.location.href);
  let likeCount = getLikeCountFromButton() || null;

  let response = await fetch(
    `${apiUrl}/votes?videoId=${videoId}&likeCount=${likeCount || ""}`,
    {
      method: "GET",
      headers: {
        Accept: "application/json",
      },
    }
  )
    .then((response) => {
      if (!response.ok) displayError(response.error);
      return response;
    })
    .then((response) => response.json())
    .catch(displayError);
  cLog("response from api:");
  cLog(JSON.stringify(response));
  if (response !== undefined && !("traceId" in response) && !statsSet) {
    processResponse(response, storedData);
  }
}

function setInitialState() {
  setState(storedData);
}

function initExtConfig() {
  initializeDisableVoteSubmission();
  initializeColoredThumbs();
  initializeColoredBar();
  initializeColorTheme();
  initializeNumberDisplayFormat();
  initializeTooltipPercentage();
  initializeTooltipPercentageMode();
  initializeNumberDisplayReformatLikes();
}

function initializeDisableVoteSubmission() {
  getBrowser().storage.sync.get(["disableVoteSubmission"], (res) => {
    if (res.disableVoteSubmission === undefined) {
      getBrowser().storage.sync.set({ disableVoteSubmission: false });
    } else {
      extConfig.disableVoteSubmission = res.disableVoteSubmission;
    }
  });
}

function initializeColoredThumbs() {
  getBrowser().storage.sync.get(["coloredThumbs"], (res) => {
    if (res.coloredThumbs === undefined) {
      getBrowser().storage.sync.set({ coloredThumbs: false });
    } else {
      extConfig.coloredThumbs = res.coloredThumbs;
    }
  });
}

function initializeColoredBar() {
  getBrowser().storage.sync.get(["coloredBar"], (res) => {
    if (res.coloredBar === undefined) {
      getBrowser().storage.sync.set({ coloredBar: false });
    } else {
      extConfig.coloredBar = res.coloredBar;
    }
  });
}

function initializeColorTheme() {
  getBrowser().storage.sync.get(["colorTheme"], (res) => {
    if (res.colorTheme === undefined) {
      getBrowser().storage.sync.set({ colorTheme: false });
    } else {
      extConfig.colorTheme = res.colorTheme;
    }
  });
}

function initializeNumberDisplayFormat() {
  getBrowser().storage.sync.get(["numberDisplayFormat"], (res) => {
    if (res.numberDisplayFormat === undefined) {
      getBrowser().storage.sync.set({ numberDisplayFormat: "compactShort" });
    } else {
      extConfig.numberDisplayFormat = res.numberDisplayFormat;
    }
  });
}

function initializeTooltipPercentage() {
  getBrowser().storage.sync.get(["showTooltipPercentage"], (res) => {
    if (res.showTooltipPercentage === undefined) {
      getBrowser().storage.sync.set({ showTooltipPercentage: false });
    } else {
      extConfig.showTooltipPercentage = res.showTooltipPercentage;
    }
  });
}

function initializeTooltipPercentageMode() {
  getBrowser().storage.sync.get(["tooltipPercentageMode"], (res) => {
    if (res.tooltipPercentageMode === undefined) {
      getBrowser().storage.sync.set({ tooltipPercentageMode: "dash_like" });
    } else {
      extConfig.tooltipPercentageMode = res.tooltipPercentageMode;
    }
  });
}

function initializeNumberDisplayReformatLikes() {
  getBrowser().storage.sync.get(["numberDisplayReformatLikes"], (res) => {
    if (res.numberDisplayReformatLikes === undefined) {
      getBrowser().storage.sync.set({ numberDisplayReformatLikes: false });
    } else {
      extConfig.numberDisplayReformatLikes = res.numberDisplayReformatLikes;
    }
  });
}

export {
  isMobile,
  isShorts,
  isVideoDisliked,
  isVideoLiked,
  isNewDesign,
  isRoundedDesign,
  getState,
  setState,
  setInitialState,
  setLikes,
  setDislikes,
  getLikeCountFromButton,
  LIKED_STATE,
  DISLIKED_STATE,
  NEUTRAL_STATE,
  extConfig,
  initExtConfig,
  storedData,
  isLikesDisabled,
};
