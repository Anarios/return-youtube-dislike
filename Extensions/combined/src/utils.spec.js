/**
 * @jest-environment jsdom
 */

import {
  __Rewire__ as rewiredUtils,
  numberFormat,
  getNumberFormatter,
  localize,
  getBrowser,
  getVideoId,
  isInViewport,
  isVideoLoaded,
  cLog,
  getColorFromTheme,
} from "./utils";
import { extConfig } from "./state";

jest.mock("./state");

describe("numberFormat", () => {
  it("should format high numbers correctly", () => {
    rewiredUtils(
      "getNumberFormatter",
      jest.fn().mockReturnValue(
        Intl.NumberFormat("en", {
          notation: "compact",
          compactDisplay: "short",
        }),
      ),
    );

    expect(numberFormat(91492)).toBe("91K");
    jest.resetAllMocks();
  });

  it("should not format input when < 1000", () => {
    rewiredUtils(
      "getNumberFormatter",
      jest.fn().mockReturnValue(
        Intl.NumberFormat("en", {
          notation: "compact",
          compactDisplay: "short",
        }),
      ),
    );

    expect(numberFormat(100)).toBe("100");
  });

  it("should not round down when rounding is diabled in config", () => {
    rewiredUtils(
      "getNumberFormatter",
      jest.fn().mockReturnValue(
        Intl.NumberFormat("de", {
          notation: "compact",
          compactDisplay: "short",
        }),
      ),
    );
    extConfig.numberDisplayRoundDown = false;

    expect(numberFormat(912391)).toBe("912.391");
  });
});

describe("getNumberFormatter", () => {
  it("should return a correct formatter with compactLong option", () => {
    expect(getNumberFormatter("compactLong")).toMatchSnapshot();
  });

  it("should return a correct formatter whith standard option", () => {
    expect(getNumberFormatter("standard")).toMatchSnapshot();
  });

  it("should return a correct formatter when using chrome", () => {
    Object.defineProperty(document.documentElement, "lang", {
      value: "en",
      configurable: true,
    });
    expect(getNumberFormatter()).toMatchSnapshot();
  });

  it("should return a correct formatter when using firefox", () => {
    Object.defineProperty(document.documentElement, "lang", {
      value: null,
      configurable: true,
    });
    expect(getNumberFormatter()).toMatchSnapshot();
  });

  it("should return a correct formatter when using the URL locale (possibly buggy)", () => {
    // Disclaimer: The case we are testing here is actually not correct and
    // am almost sure I found a bug in the actual implementation of
    // getNumberFormatter(). As I am currently writing acceptance tests only
    // e.g. tests that are exclusively testing the current behaviour as is, as
    // a first step of adding tests to the codebase, I will add it like this
    // for now and document how we should actually be testing down below.
    // The reason why this bug is not causing faulty behaviour is, that we
    // have a fallback for it, that is always triggered.

    // This is how we make the test go green:
    Object.defineProperty(document.documentElement, "lang", {
      value: null,
      configurable: true,
    });
    Object.defineProperty(navigator, "language", {
      value: null,
      configurable: true,
    });
    const mockedNode = document.createElement("link");

    mockedNode.setAttribute("href", "https://www.youtube.com/opensearch?locale=en");

    document.querySelectorAll = jest.fn().mockReturnValue([mockedNode]);
    expect(getNumberFormatter()).toMatchSnapshot();
  });

  it.skip("should return a correct formatter when using the URL locale", () => {
    // But we actually want to test like so, which is the correct value of the locale attribute:
    Object.defineProperty(document.documentElement, "lang", {
      value: null,
      configurable: true,
    });
    Object.defineProperty(navigator, "language", {
      value: null,
      configurable: true,
    });
    const mockedNode = document.createElement("link");
    mockedNode.setAttribute("href", "https://www.youtube.com/opensearch?locale=en_US");

    document.querySelectorAll = jest.fn().mockReturnValue([mockedNode]);
    expect(getNumberFormatter()).toMatchSnapshot();
  });

  it("should return a correct formatter when falling back to default locale", () => {
    Object.defineProperty(document.documentElement, "lang", {
      value: null,
      configurable: true,
    });
    Object.defineProperty(navigator, "language", {
      value: null,
      configurable: true,
    });
    expect(getNumberFormatter()).toMatchSnapshot();
  });

  describe("localize", () => {
    it("should return a translated string", () => {
      global.chrome = {
        i18n: {
          getMessage: () => "Return YouTube Dislike",
        },
      };
      expect(localize("extensionName")).toBe("Return YouTube Dislike");
    });
  });

  describe("getBrowser", () => {
    it("should return a chrome browser instance", () => {
      global.chrome = {
        runtime: jest.fn(),
      };
      expect(getBrowser()).toBeDefined();
    });

    it("should return a mozilla browser instance", () => {
      global.chrome.runtime = undefined;
      global.browser = {
        runtime: jest.fn(),
      };
      expect(getBrowser()).toBeDefined();
    });

    it("should fail on unsupported browser", () => {
      global.chrome = undefined;
      global.browser = undefined;

      const log = console.log;
      console.log = jest.fn();

      expect(getBrowser()).toBe(false);
      expect(console.log).toBeCalledWith("browser is not supported");

      console.log = log;
    });
  });
  describe("getVideoId", () => {
    it("should return the video Id on valid url input", () => {
      const url = "https://www.youtube.com/watch?v=dQw4w9WgXcQ";
      expect(getVideoId(url)).toBe("dQw4w9WgXcQ");
    });

    it("should return the video Id on valid shorts url input", () => {
      const url = "https://www.youtube.com/shorts/niMcQ9Vmmj8";
      expect(getVideoId(url)).toBe("niMcQ9Vmmj8");
    });

    it("should return the video Id on clip url input", () => {
      document.querySelector = jest.fn().mockReturnValue({
        content: "niMcQ9Vmmj8",
      });
      const url = "https://www.youtube.com/clip";
      expect(getVideoId(url)).toBe("niMcQ9Vmmj8");
    });

    it("should return null on invalid url input", () => {
      const url = "https://www.youtube.com/wrong-url";
      expect(getVideoId(url)).toBe(null);
    });
  });
  describe("isInViewport", () => {
    it("should return true if element is in viewport", () => {
      const rect = {
        getBoundingClientRect: jest.fn().mockReturnValue({
          top: 0,
          left: 0,
          bottom: 0,
          right: 0,
        }),
      };
      expect(isInViewport(rect)).toBe(false);
    });

    it("should return false if element is out of viewport", () => {
      const rect = {
        getBoundingClientRect: jest.fn().mockReturnValue({
          top: 0,
          left: 0,
          bottom: 769,
          right: 0,
        }),
      };
      expect(isInViewport(rect)).toBe(false);
    });
  });
  describe("isVideoLoaded", () => {
    describe("on desktop", () => {
      it("should return true on loaded video", () => {
        rewiredUtils("getVideoId", jest.fn().mockReturnValue("fakeId"));
        document.querySelector = jest.fn().mockReturnValue("notNull");
        expect(isVideoLoaded()).toBe(true);
        expect(document.querySelector).toBeCalledWith("ytd-watch-grid[video-id='fakeId']");
      });

      it("should return false on not loaded video", () => {
        rewiredUtils("getVideoId", jest.fn().mockReturnValue("fakeId"));
        document.querySelector = jest.fn().mockReturnValue(null);
        expect(isVideoLoaded()).toBe(false);
        expect(document.querySelector).toBeCalledWith("ytd-watch-flexy[video-id='fakeId']");
      });
    });

    describe("on mobile", () => {
      it("should return true on loaded video", () => {
        rewiredUtils("getVideoId", jest.fn().mockReturnValue("fakeId"));
        document.querySelector = jest.fn().mockReturnValueOnce(null).mockReturnValue("notNull");
        expect(isVideoLoaded()).toBe(true);
        expect(document.querySelector).toHaveBeenLastCalledWith("ytd-watch-flexy[video-id='fakeId']");
      });

      it("should return false on not loaded video", () => {
        rewiredUtils("getVideoId", jest.fn().mockReturnValue("fakeId"));
        document.querySelector = jest.fn().mockReturnValueOnce(null).mockReturnValue(null);
        expect(isVideoLoaded()).toBe(false);
        expect(document.querySelector).toHaveBeenLastCalledWith('#player[loading="false"]:not([hidden])');
      });
    });
  });
  describe("cLog", () => {
    it("should log a message with the correct prefix", () => {
      const log = console.log;
      console.log = jest.fn();
      cLog("Test message");

      expect(console.log).toBeCalledWith("[return youtube dislike]: Test message");

      console.log = log;
    });

    it("should log a message with the correct prefix when given a writer", () => {
      const fakeWriter = jest.fn();

      cLog("Test message", fakeWriter);
      expect(fakeWriter).toBeCalledWith("[return youtube dislike]: Test message");
    });
  });
  describe("getColorFromTheme", () => {
    describe("accessible", () => {
      it("should return the correct color for like", () => {
        extConfig.colorTheme = "accessible";
        expect(getColorFromTheme(true)).toBe("dodgerblue");
      });

      it("should return the correct color for not liked", () => {
        extConfig.colorTheme = "accessible";
        expect(getColorFromTheme(false)).toBe("gold");
      });
    });

    describe("neon", () => {
      it("should return the correct color for like", () => {
        extConfig.colorTheme = "neon";
        expect(getColorFromTheme(true)).toBe("aqua");
      });

      it("should return the correct color for not liked", () => {
        extConfig.colorTheme = "neon";
        expect(getColorFromTheme(false)).toBe("magenta");
      });
    });

    describe("classic", () => {
      it("should return the correct color for like", () => {
        extConfig.colorTheme = "classic";
        expect(getColorFromTheme(true)).toBe("lime");
      });

      it("should return the correct color for not liked", () => {
        extConfig.colorTheme = "classic";
        expect(getColorFromTheme(false)).toBe("red");
      });
    });
  });
});
