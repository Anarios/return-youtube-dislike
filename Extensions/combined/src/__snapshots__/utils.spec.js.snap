// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNumberFormatter should return a correct formatter when falling back to default locale 1`] = `NumberFormat {}`;

exports[`getNumberFormatter should return a correct formatter when using chrome 1`] = `NumberFormat {}`;

exports[`getNumberFormatter should return a correct formatter when using firefox 1`] = `NumberFormat {}`;

exports[`getNumberFormatter should return a correct formatter when using the URL locale (possibly buggy) 1`] = `NumberFormat {}`;

exports[`getNumberFormatter should return a correct formatter when using the URL locale 1`] = `NumberFormat {}`;

exports[`getNumberFormatter should return a correct formatter whith standard option 1`] = `NumberFormat {}`;

exports[`getNumberFormatter should return a correct formatter with compactLong option 1`] = `NumberFormat {}`;
